src/extent.c

```diff
@@ -508,49 +562,36 @@ extent_activate_locked(tsdn_t *tsdn, arena_t *arena, extents_t *extents,
 }
 
 static bool
-extent_rtree_acquire(tsdn_t *tsdn, rtree_ctx_t *rtree_ctx,
+extent_rtree_leaf_elms_lookup(tsdn_t *tsdn, rtree_ctx_t *rtree_ctx,
     const extent_t *extent, bool dependent, bool init_missing,
     rtree_leaf_elm_t **r_elm_a, rtree_leaf_elm_t **r_elm_b) {
-	*r_elm_a = rtree_leaf_elm_acquire(tsdn, &extents_rtree, rtree_ctx,
+	*r_elm_a = NULL;
```

Comment:
Redundant initialization is brittle in my experience, and almost never necessary.
```diff
diff --git a/src/extent.c b/src/extent.c
index 5c8b699..9b2e715 100644
--- a/src/extent.c
+++ b/src/extent.c
@@ -565,9 +565,6 @@ static bool
 extent_rtree_leaf_elms_lookup(tsdn_t *tsdn, rtree_ctx_t *rtree_ctx,
     const extent_t *extent, bool dependent, bool init_missing,
     rtree_leaf_elm_t **r_elm_a, rtree_leaf_elm_t **r_elm_b) {
-       *r_elm_a = NULL;
-       *r_elm_b = NULL;
-
        *r_elm_a = rtree_leaf_elm_lookup(tsdn, &extents_rtree, rtree_ctx,
            (uintptr_t)extent_base_get(extent), dependent, init_missing);
        if (!dependent && *r_elm_a == NULL) {
@@ -580,7 +577,7 @@ extent_rtree_leaf_elms_lookup(tsdn_t *tsdn, rtree_ctx_t *rtree_ctx,
        if (!dependent && *r_elm_b == NULL) {
                return true;
        }
-       assert(*r_elm_a != NULL);
+       assert(*r_elm_b != NULL);
 
        return false;
 }
```